src/Geometry.cpp

```diff
@@ -1081,6 +1105,10 @@ bool Geometry::withinBounds(LocalCoords* coords){
 }
 
 
+void Geometry::setFSRCentroid(int fsr, Point* centroid){
```

Comment:
I like that compromise, but it might be wise to consider a cleaner solution in the future.